FinancialTracker/DatabaseManager.swift

```diff
@@ -0,0 +1,226 @@
+//
+//  DatabaseManager.swift
+//  FinancialTracker
+//
+//  Created by Boyan Pavlov on 5.02.22.
+//
+
+import Foundation
+import Firebase
+import FirebaseFirestore
+
+protocol DatabaseManagerDelegate: AnyObject {
+    func databaseManagerDidUserChange(sender: DatabaseManager)
+}
+
+class DatabaseManager {
+    private let firestore = Firestore.firestore()
+    private let delegatesCollection = DelegatesCollection<DatabaseManagerDelegate>()
+ 
+    private(set) var currentUser: User?
+    private(set) var authManager: AuthManager?
+    
+    init() {}
+    
+    func setAuthManager(authManager: AuthManager) {
+        self.authManager = authManager
```

Comment:
`self.authManager = authManager` can happen in the `init` for the `DatabaseManager`

Reply:
Also, I am not seeing the `authManager` initialised within `DatabaseManager` actually being used within `DatabaseManager` at all. Are you sure you need it?

Reply:
It is used in the signOut function every controller has